src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.updateAliveList();
+    }
+  }
```

Comment:
[CODE QUALITY] The `checkHealth` method in the `Animal` class introduces tight coupling between the animal's health check and the static method `updateAliveList`. Instead of calling `updateAliveList` directly, consider using event-driven communication or a different approach that decouples the health check from the list update. This will improve the maintainability and flexibility of the code. For example, you could have a method that is called after the `health` property changes that takes care of any necessary updates.